naptime-graphql/src/test/scala/org/coursera/naptime/ari/graphql/helpers/ArgumentBuilder.scala

```diff
@@ -0,0 +1,43 @@
+package org.coursera.naptime.ari.graphql.helpers
+
+import org.coursera.naptime.PaginationConfiguration
+import sangria.schema.Args
+import sangria.schema.Argument
+import sangria.schema.IntType
+import sangria.schema.OptionInputType
+import sangria.schema.StringType
+
+import scala.collection.concurrent.TrieMap
+
+object ArgumentBuilder {
+  def buildArgs(argumentDefinitions: List[Argument[_]], argumentInputs: Map[String, Any]): Args = {
+    val argsWithDefault = argumentDefinitions.filter(_.defaultValue.isDefined).map(_.name).toSet
+    val optionalArgs = argumentDefinitions.filter(_.inputValueType.isOptional).map(_.name).toSet
+    val undefinedArgs = argumentDefinitions.map(_.name)
+      .filterNot(argsWithDefault.contains)
+      .filterNot(optionalArgs.contains)
+      .filterNot(argumentInputs.contains)
+      .toSet
+    val defaultInfo = argumentDefinitions
```

Comment:
i _think_ you can combine the filter and map into a `collect` statement:
```scala
argumentDefinitions.collect {
  case definition if argsWithDefault.contains(definition.name) => definition.name -> definition.defaultValue.get._1
}
```

see http://ochafik.com/p_393 (or google) for more info